sdcm/cluster.py

```diff
@@ -110,6 +110,8 @@ class Setup:
     INTRA_NODE_COMM_PUBLIC = False
     RSYSLOG_ADDRESS = None
     DECODING_QUEUE = None
+    USE_LEGACY_CLUSTER_INIT = False
+    AUTO_BOOTSTRAP = True
```

Comment:
We should be removing global variables, not adding them :(

Imagine we want to have two clusters, one bootstrapped using the "legacy method" and the other using the proper method. We set `USE_LEGACY_CLUSTER_INIT = True` here. Which cluster does this affect? One of them? Which one? Or all of them?

IMO this should be configurable per cluster, hence it should be passed as a parameter to the cluster object.

Reply:
@kbr- 
1) Setup is a Singleton object that is needed in different stages of SCT run to understand the test params or a state
2) Our support for multiple DB clusters in the SCT is very limited (to Gemini test), so by definition, all test run parameters in SCT are global for all clusters

Reply:
@bentsi is there no hope for more complex cluster setups in the future (multiple clusters in non-gemini tests) etc.?

Also, I don't understand why we need the `Setup` singleton. After all, all the code already has access to `self.params` which contains the config.

Reply:
When we will have a complex setup we will develop support for them. It's not a requirement that we even have task in our backlog. `Setup` doesn't only have values from params object but other like test_dir, test_id, test_name, type of the connectivity between nodes, etc.
`self.params` is a legacy structure that represents the configuration that we get from the test YAML/s. 
We will unify it with Setup in future

Reply:
It's good that you want to unify the different places where config is stored. I agree. So it would be nice to have a single `Setup` object, and not also `self.params`.

However, I think you should not go into a direction where everything is stored in a singleton, which is basically a bunch of global variables ([singleton anti-pattern](https://www.michaelsafyan.com/tech/design/patterns/singleton)). It will make any future development much harder. And the alternatives to using a singleton are not at all harder to use, so why use a singleton?

Reply:
@kbr- thanks for pointing me out to the article. We will take this into consideration when we will refactor the Setup class